src/connection.rs

```diff
@@ -188,6 +200,24 @@ impl Connection {
 
         Ok(())
     }
+
+    fn poll_message(&mut self, cx: &mut Context<'_>) -> Poll<Option<SubstreamMessage>> {
+        loop {
```

Comment:
why is there a loop here? there shouldn't be any infinite loops inside a `poll` function afaik, it seems unnecessary to have this logically